superset-frontend/src/components/ErrorBoundary/index.tsx

```diff
@@ -17,27 +17,30 @@
  * under the License.
  */
 import React from 'react';
-import PropTypes from 'prop-types';
 import { t } from '@superset-ui/core';
 import ErrorMessageWithStackTrace from 'src/components/ErrorMessage/ErrorMessageWithStackTrace';
 
-const propTypes = {
-  children: PropTypes.node.isRequired,
-  onError: PropTypes.func,
-  showMessage: PropTypes.bool,
-};
-const defaultProps = {
-  onError: () => {},
-  showMessage: true,
-};
+interface ErrorBoundaryProps {
+  children: React.ReactNode;
+  onError?: Function;
+  showMessage?: boolean;
+}
+
+export default class ErrorBoundary extends React.Component<
+  ErrorBoundaryProps,
+  { error: Error | null; info: React.ErrorInfo | null }
```

Comment:
Maybe let's make this into an `ErrorBoundaryState` too?

Reply:
Yessir!